### PR TITLE
Import POWHEG r3693

### PR DESCRIPTION
**Pull request by mfasDa**

### CODE REVIEW

**Comment by mfasDa** [comment]
After merging into master a branch alice/r3693 should be created, and from this a tag r3693-alice should be created. The version 3178 used by ALICE up to now is > 4 years old and i.e. doesn't yet support LHAPDF6 supporting more modern PDF sets.

**Comment by maireiphc** [comment]
Hi Markus,

  is the fix for dijet/Makefile still needed or conversely still in, after your complete refurbishment with this new POWHEG version ? (Jochen fix for previous version 29 mars 2017 : https://github.com/alisw/POWHEG/pull/2/commits)/39db397a4dd408d418b9925a37e5d94e0e30d961)

**Comment by mfasDa** [comment]
Hi @maireiphc , for me it compiled and ran without the fix, but probably yes, I have seen similar issues with SHERPA. It should be ported as well. Let me prepare the patch. I will keep it as a separate commit.

**Comment by maireiphc** [comment]
Hi @mfasDa (adding @dstocco as well, as author of the 2016 mods on Powheg)

pushing further down, I did not check whether even earlier commits from Diego about nPDF inclusion have made their way through the official POWHEG release or if this is ALICE specific business, i.e. that should also be of concern here and thus patched to the revision r3693 we plan to have.
https://github.com/alisw/POWHEG/commits/alice/r3178

(I am waiting for a merge of a new Rivet tag to master. This should enable AliGenerators build checks to succeed again and thus be valuable for the tests here.
https://github.com/alisw/alidist/pull/2080
)

**Comment by dstocco** [comment]
Hi @maireiphc ,
sorry, I totally overlooked this thread.
The modifications that I did were meant to be able to use nPDF. This was needed because the nPDFs were tabulated as a kind of RAA to be applied to free-parton PDFs.
However, if I understand correctly, in LHAPDF v6 the nPDFs are provided as "standard" PDF sets, so one just needs to use them instead of the free PDFs, without further hacks.

However, I also implemented an hack to be able to use neutron PDFs instead of proton ones:
https://github.com/alisw/POWHEG/pull/2/commits/5ac88679ec9f25e50557c43daf3574098e3f788b
The hack is rather simple: in case of neutrons we swap the role of quark u and d. This is relevant for EW boson simulations in heavy=ion collisions. This is an ALICE hack (given my poor fortran skills I have never asked to add this in the official POWHEG package), so it will have to be re-implemented if we want to be able to keep this feature.

We can leave things as they are right now and implement this feature in the future if needed.

Cheers,
Diego

**Review by maireiphc — APPROVED** [review]

**Comment by maireiphc** [comment]
Ok ,thanks @dstocco for your explanations.

**Comment by maireiphc** [comment]
@ktf it seems that PAG-MC does not own POWHEG.
https://github.com/orgs/alisw/teams/pagmc/repositories

Checking into alidist, POWHEG is just there for AliGenerators. I do not see it elsewhere in recipes.
Should PAGMC become the admin of the code ?

**Comment by qgp** [comment]
@maireiphc @mfasDa Following the alisw conventions, only the upstream code should go to the master branch, all the patches should then go to the respective alice/v... branch. At least, this is what I was asked to adhere to in the past by alisw maintainers. Did this strategy change?

**Comment by maireiphc** [comment]
@mfasDa, @qgp 

How should we proceed : 
- merge or squash-merge this to master ?
- create a new branch alice/r3693 ?
- tag a new release r3693-alice1 ?

I am bit puzzled to see 8 commits there, i.e. also the past editions on r3178. Can we safely merge all these commits ? In particular, there are ALICE specific commits that I would have expected to be ported only once the alice branch is in place ?

In other words, if the old strategy still holds, it means that only alisw code admin can bump source from the official release to master and then let outsider player like Markus, edit and commit to adapt things to ALICE.
Is that correct ?

**Comment by mfasDa** [comment]
Hi @qgp , I am not sure but it might be that my PR also includes the patch for dijet. Maybe do this in a clean way:
- open new PR  with the original sources.
- open a branch for the alice patches
- Make PR into the patch branch for the makefile fix
- Make POWHEG tag

**Comment by qgp** [comment]
> * open new PR  with the original sources.
> * open a branch for the alice patches
> * Make PR into the patch branch for the makefile fix
> * Make POWHEG tag

Hi @mfasDa, this would be the cleanest approach. Alternatively, you could at least make this PR against a new branch alice/r3693.

**Comment by maireiphc** [comment]
Ok then there are 2 questions for the alisw admins (@ktf, @sawenzel) :
1. Who has to be admin on this alisw/POWHEG repo : alisw or pagmc ?
2. Is the historical development policy still enforced :

-  master = clean copy of the official sources including potential bugs 
- then, anything like further dev or ALICE patch must go into a branch alice/vX.Y.Z branch from where we draft our release

**Comment by mfasDa** [comment]
I am closing the PR in favour of #3, separating the PR into original sources and ALICE patches.
